README

```diff
@@ -1,43 +1,79 @@
+--------
+opa-hfi1
+--------
+This is the source version of the hfi1 driver that is released with official IFS
+builds. However it does at times, depending on the branch contain additional
+fixes that could not make the IFS release that customers may find useful.
 
+Branches are generally named based on IFS release. For instance:
+opa-10_11_0_1 represents IFS release 10.11.0.1
+
+
+---------------------
 Building the software
+---------------------
 
-./do-update-makerpm.sh -S ${PWD} -w ${PWD}/tmp
+Pre-reqs
+--------
+Install necessary build tool chain and kernel source as well as kernel
+development packages to be able to build out of tree kernel modules. For support
+with this, contact your Linux distro.
+
+Before continuing it is highly recommended to note down the srcversion of the
+driver you are currently running.
+
+cat /sys/module/hfi1/srcversion
 
-To build with vnic, use the command
-Note: at this time, this is only operational on RHEL 7.5
+Note this is available in the modinfo command but modinfo only tells you the
+default driver's information, by default. It is not necessarily the driver that 
+is currently running.
 
-./do-update-makerpm.sh -V -S ${PWD} -w ${PWD}/tmp
+Also be sure to clear out the tmp directory before rebuilding if this isn't the
+first time you are trying to build.
+
```

Comment:
I think there should be an outline of the steps required, before diving in to building. It's not intuitive to me that one needs to build a source rpm and then extract that and build a binary rpm. It might be helpful to see a complete list of steps, all the way through the modprobe/insmod.

Reply:
Take a look at the version I just pushed. See if that helps.